docs/api.md

```diff
@@ -520,7 +527,20 @@ The Metal3MachineTemplate contains following two specification fields:
 
 ### Enabling nodeReuse feature
 
-This feature can be desirable and enabled in scenarios such as upgrade/remediation, where root and externally attached disks of the BareMetalHosts needs to be left untouched and same pool of BareMetalHosts reused during the re-provisioning. To achieve that, `nodeReuse` field in `Metal3MachineTemplateSpec` must be set to `True`, and next CAPM3 Machine controller:
+This feature can be desirable and enabled in scenarios such as upgrade or node remediation. For example, the same pool of hosts need to be used after cluster upgrade and no data of secondary storage should be lost.
+To achieve that:
+
+1. `spec.nodeReuse` field of metal3MachineTemplate must be set to `True`. This tells that we want to reuse the same hosts
+  after the upgrade, or to be exact same baremetalHosts should be provisioned.
+
+1. `spec.template.spec.automatedCleaningMode` field of metal3MachineTemplate must be set to `disabled`. This tells that we
```

Comment:
We should also update [example M3MT](https://github.com/metal3-io/cluster-api-provider-metal3/pull/156/files#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abR559) with default value of `automatedCleaningMode` field.

Reply:
Sorry, it seems URL is redirecting me to a commit, where shall I update it, in the document?

Reply:
Lines 548 (Example Metal3MachineTemplate) and onwards, bit down on this same document

Reply:
There is no spec.template part. Not sure if I should add a whole spec of Metal3Machine for showing cleaning field. Maybe only Metal3Machine example with `AutomatedCleaningMode` is enough?

Reply:
I am not sure if I got your point. I was suggesting to put `AutomatedCleaningMode` field in example Metal3MachineTemplate as we did for nodeReuse field, i.e: 

```
      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4 
      kind: Metal3MachineTemplate
      metadata:
      name: md-0
      spec:
        nodeReuse: false
        template:
          spec:
            AutomatedCleaningMode: metadata
            image:
              url: https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img
              checksum: https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img.md5sum
            hostSelector:
              matchLabels:
                key1: value1
              matchExpressions:
                key: key2
                operator: in
                values: {‘abc’, ‘123’, ‘value2’}
            dataTemplate:
              Name: md-0-metadata
```

Reply:
🤦‍♂️, fixing it